protocols/manage/propagate.go

```diff
@@ -0,0 +1,199 @@
+package manage
+
+import (
+	"sync"
+
+	"time"
+
+	"reflect"
+
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+func init() {
+	sda.ProtocolRegisterName("Propagate", NewPropagateProtocol)
+}
+
+// Propagate is a protocol that sends some data to all attached nodes
+// and waits for confirmation before returning.
+type Propagate struct {
+	*sda.TreeNodeInstance
+	onData    func(network.ProtocolMessage)
+	onDoneCb  func(int)
+	sd        *PropagateSendData
+	ChannelSD chan struct {
+		*sda.TreeNode
+		PropagateSendData
+	}
+	ChannelReply chan struct {
+		*sda.TreeNode
+		PropagateReply
+	}
+
+	received int
+	subtree  int
+	sync.Mutex
+}
+
+// CreateProtocolEntity is the necessary interface to start a protocol.
+// It is implemented by Service and Overlay.
+type CreateProtocolEntity interface {
+	CreateProtocol(t *sda.Tree, name string) (sda.ProtocolInstance, error)
+	Entity() *network.Entity
+}
+
+// PropagateSendData is the message to pass the data to the children
+type PropagateSendData struct {
+	// Data is the data to transmit
+	Data []byte
+	// Msec is the timeout in milliseconds
+	Msec int
```

Comment:
Yes, if I read the code I can see that. But if the comment only says "the timeout" it's not really useful. I would either remove the comment (than it's clear that one must read the code to find out what will happen with `Msec` or explain which timeout so that one doesn't need to read through the code).